```
  /var/tmp/vite-plus-20260412-8353-e73mmv/vite-plus-0.1.16/node_modules/.pnpm/@ast-grep+napi@0.40.4/node_modules/@ast-grep/napi/index.js:385
      throw new Error(
            ^
  
  Error: Cannot find native binding. npm has a bug related to optional dependencies (https://github.com/npm/cli/issues/4828). Please try `npm i` again after removing both package-lock.json and node_modules directory.
      at Object.<anonymous> (/var/tmp/vite-plus-20260412-8353-e73mmv/vite-plus-0.1.16/node_modules/.pnpm/@ast-grep+napi@0.40.4/node_modules/@ast-grep/napi/index.js:385:11)
      at Module._compile (node:internal/modules/cjs/loader:1829:14)
      at Module._extensions..js (node:internal/modules/cjs/loader:1969:10)
      at Object.newLoader [as .js] (/var/tmp/vite-plus-20260412-8353-e73mmv/vite-plus-0.1.16/node_modules/.pnpm/pirates@4.0.7/node_modules/pirates/lib/index.js:134:7)
      at Module.load (node:internal/modules/cjs/loader:1552:32)
      at Module._load (node:internal/modules/cjs/loader:1354:12)
      at wrapModuleLoad (node:internal/modules/cjs/loader:255:19)
      at loadCJSModuleWithModuleLoad (node:internal/modules/esm/translators:327:3)
      at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:232:7)
      at ModuleJob.run (node:internal/modules/esm/module_job:437:25) {
    [cause]: [
      Error: Cannot find module './ast-grep-napi.linux-arm64-gnu.node'
      Require stack:
      - /var/tmp/vite-plus-20260412-8353-e73mmv/vite-plus-0.1.16/node_modules/.pnpm/@ast-grep+napi@0.40.4/node_modules/@ast-grep/napi/index.js
          at Module._resolveFilename (node:internal/modules/cjs/loader:1475:15)
          at wrapResolveFilename (node:internal/modules/cjs/loader:1048:27)
          at defaultResolveImplForCJSLoading (node:internal/modules/cjs/loader:1072:10)
          at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1093:12)
          at Module._load (node:internal/modules/cjs/loader:1261:25)
          at wrapModuleLoad (node:internal/modules/cjs/loader:255:19)
          at Module.require (node:internal/modules/cjs/loader:1575:12)
          at require (node:internal/modules/helpers:191:16)
          at requireNative (/var/tmp/vite-plus-20260412-8353-e73mmv/vite-plus-0.1.16/node_modules/.pnpm/@ast-grep+napi@0.40.4/node_modules/@ast-grep/napi/index.js:237:18)
          at Object.<anonymous> (/var/tmp/vite-plus-20260412-8353-e73mmv/vite-plus-0.1.16/node_modules/.pnpm/@ast-grep+napi@0.40.4/node_modules/@ast-grep/napi/index.js:362:17) {
        code: 'MODULE_NOT_FOUND',
        requireStack: [
          '/var/tmp/vite-plus-20260412-8353-e73mmv/vite-plus-0.1.16/node_modules/.pnpm/@ast-grep+napi@0.40.4/node_modules/@ast-grep/napi/index.js'
        ]
      },
      Error: /var/tmp/vite-plus-20260412-8353-e73mmv/vite-plus-0.1.16/node_modules/.pnpm/@ast-grep+napi-linux-arm64-gnu@0.40.4/node_modules/@ast-grep/napi-linux-arm64-gnu/ast-grep-napi.linux-arm64-gnu.node: undefined symbol: le16toh
          at Module._extensions..node (node:internal/modules/cjs/loader:1996:18)
          at Module.load (node:internal/modules/cjs/loader:1552:32)
          at Module._load (node:internal/modules/cjs/loader:1354:12)
          at wrapModuleLoad (node:internal/modules/cjs/loader:255:19)
          at Module.require (node:internal/modules/cjs/loader:1575:12)
          at require (node:internal/modules/helpers:191:16)
          at requireNative (/var/tmp/vite-plus-20260412-8353-e73mmv/vite-plus-0.1.16/node_modules/.pnpm/@ast-grep+napi@0.40.4/node_modules/@ast-grep/napi/index.js:242:18)
          at Object.<anonymous> (/var/tmp/vite-plus-20260412-8353-e73mmv/vite-plus-0.1.16/node_modules/.pnpm/@ast-grep+napi@0.40.4/node_modules/@ast-grep/napi/index.js:362:17)
          at Module._compile (node:internal/modules/cjs/loader:1829:14)
          at Module._extensions..js (node:internal/modules/cjs/loader:1969:10) {
        code: 'ERR_DLOPEN_FAILED'
      }
    ]
  }
  
  Node.js v25.9.0
  /var/tmp/vite-plus-20260412-8353-e73mmv/vite-plus-0.1.16/packages/core:
   ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @voidzero-dev/vite-plus-core@0.0.0 build: `oxnode -C dev ./build.ts`
  Exit status 1
  error: Recipe `build` failed on line 31 with exit code 1
```

--- a/Formula/v/vite-plus.rb
+++ b/Formula/v/vite-plus.rb
@@ -1,0 +1,105 @@
+class VitePlus < Formula
+  desc "Vite+ is the unified toolchain and entry point for web development"
+  homepage "https://viteplus.dev"
+  url "https://github.com/voidzero-dev/vite-plus/archive/refs/tags/v0.1.18.tar.gz"
+  sha256 "6ee77f933315145b50cf21db8505f6175fd744b4fac78c7d275562508097cbc0"
+  license "MIT"
+  head "https://github.com/voidzero-dev/vite-plus.git", branch: "main"
+
+  depends_on "cmake" => :build
+  depends_on "just" => :build
+  depends_on "pnpm" => :build
+  depends_on "rust" => :build
+  depends_on "node"
+
+  resource "rolldown" do
+    url "https://github.com/rolldown/rolldown.git",
+        revision: "27cb729813ae7facfe106a859458c199cc19955e"
+    version "27cb729813ae7facfe106a859458c199cc19955e"
+
+    livecheck do
+      url "https://raw.githubusercontent.com/voidzero-dev/vite-plus/refs/tags/v#{LATEST_VERSION}/packages/tools/.upstream-versions.json"
+      strategy :json do |json|
+        json.dig("rolldown", "hash")
+      end
+    end
+  end
+
+  resource "vite" do
+    url "https://github.com/vitejs/vite.git",
+        revision: "6e585dcb05a3b159fba7ae57f7faf0b1eca7a390"
+    version "6e585dcb05a3b159fba7ae57f7faf0b1eca7a390"
+
+    livecheck do
+      url "https://raw.githubusercontent.com/voidzero-dev/vite-plus/refs/tags/v#{LATEST_VERSION}/packages/tools/.upstream-versions.json"
+      strategy :json do |json|
+        json.dig("vite", "hash")
+      end
+    end
+  end
+
+  def install
+    resource("rolldown").stage buildpath/"rolldown"
+    resource("vite").stage buildpath/"vite"
+
+    ENV["NPM_CONFIG_MANAGE_PACKAGE_MANAGER_VERSIONS"] = "false"
+    ENV["CARGO_NET_GIT_FETCH_WITH_CLI"] = "true"
+    ENV["VITE_PLUS_VERSION"] = version.to_s
+    ENV["CARGO_PKG_VERSION"] = version.to_s
+
+    # fspy normally embeds this preload library via nightly Cargo's `-Z bindeps`.
+    # Use a stable-Rust stub to keep the CLI buildable without nightly.
+    ENV["RUSTC_BOOTSTRAP"] = "1"
+
+    (buildpath/"fspy_stub.c").write "void __fspy_stub(void) {}\n"
+    fspy_stub = buildpath/shared_library("libfspy_preload_unix")
+    system ENV.cc, OS.mac? ? "-dynamiclib" : "-shared", "-o", fspy_stub, "fspy_stub.c"
+    ENV["CARGO_CDYLIB_FILE_FSPY_PRELOAD_UNIX"] = fspy_stub
+
+    inreplace "crates/vite_global_cli/Cargo.toml", /version = "0.0.0"/, "version = \"#{version}\""
+
+    if OS.linux? && Hardware::CPU.arm?
+      # @ast-grep/napi 0.40.4 linux-arm64-gnu has undefined symbol: le16toh; 0.40.5 fixes it
+      lockfile = buildpath/"pnpm-lock.yaml"
+      inreplace lockfile,
+        "  '@ast-grep/napi-linux-arm64-gnu@0.40.4':\n    resolution: " \
+        "{integrity: " \
+        "sha512-1CeDsK6WRMz169mTXLfXdn2GkQAsMkYbqGd7mHDa2VqutJwDYrqe6t4QiFAlr+LRT2bQuExpPh3AiC8BNd6UQQ==}",
+        "  '@ast-grep/napi-linux-arm64-gnu@0.40.5':\n    resolution: " \
+        "{integrity: " \
+        "sha512-nBRCbyoS87uqkaw4Oyfe5VO+SRm2B+0g0T8ME69Qry9ShMf41a2bTdpcQx9e8scZPogq+CTwDHo3THyBV71l9w==}"
+      inreplace lockfile,
+        "  '@ast-grep/napi-linux-arm64-gnu@0.40.4':\n    optional: true",
+        "  '@ast-grep/napi-linux-arm64-gnu@0.40.5':\n    optional: true"
+      inreplace lockfile,
+        "      '@ast-grep/napi-linux-arm64-gnu': 0.40.4",
+        "      '@ast-grep/napi-linux-arm64-gnu': 0.40.5"
+    end
+
+    system "just", "build"
+    system "cargo", "install", *std_cargo_args(path: "crates/vite_global_cli")
+
+    system "pnpm", "--filter=vite-plus", "deploy", "--prod", "--legacy", "--no-optional",
+           prefix/"node_modules/vite-plus"
+    rm_r (prefix/"node_modules/vite-plus/node_modules/.pnpm").glob("*/node_modules/*/prebuilds/{darwin,ios}-x64*")
+    rm_r (prefix/"node_modules/vite-plus/node_modules/.pnpm").glob("fsevents@*/node_modules/fsevents")
+
+    # Symlink vp to vpr and vpx. These are detected at runtime by argv[0]
+    bin.install_symlink bin/"vp" => "vpr"
+    bin.install_symlink bin/"vp" => "vpx"
+
+    generate_completions_from_executable(bin/"vp", shell_parameter_format: :clap)
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/vp --version")
+
+    system bin/"vp", "create", "vite:application", "--no-interactive", "--directory", "test-app"
+    assert_path_exists testpath/"test-app/package.json"
+
+    cd testpath/"test-app" do
+      output = shell_output("#{bin}/vp fmt")
+      assert_match "Finished", output
+    end
+  end
+end

